kubetest/kops.go

```diff
@@ -172,12 +185,32 @@ func newKops(provider, gcpProject, cluster string) (*kops, error) {
 	if err := os.Setenv("AWS_SSH_KEY", sshKey); err != nil {
 		return nil, err
 	}
-	// ZONE is required by the AWS e2e tests.
-	zones := strings.Split(*kopsZones, ",")
-	if err := os.Setenv("ZONE", zones[0]); err != nil {
-		return nil, err
+
+	// zones are required by the kops e2e tests.
+	var zones []string
+
+	// if zones is set to zero and gcp project is not set then pick random aws zone
+	if *kopsZones == "" && provider == "aws" {
+		z, err := getRandomAWSZones(*kopsHA)
+		if err != nil {
+			return nil, err
+		}
+		zones = *z
```

Comment:
I'd have `getRandomAWSZones` return []string, and just check that `if len(zones) == 0) { return nil, errors.New("could not pick zones") }` or similar

Reply:
hrmmm I get what you are saying, but I would rather check for an error.  That is why I made it a pointer.  Can you live with it?